src/main/java/dev/andrylat/carsharing/dao/BodyTypeDAO.java

```diff
@@ -0,0 +1,87 @@
+package dev.andrylat.carsharing.dao;
+
+import dev.andrylat.carsharing.models.BodyType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class BodyTypeDAO {
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public BodyTypeDAO(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public List<BodyType> getAll() {
+        return jdbcTemplate.query("SELECT * FROM body_types", new BodyTypeMapper());
+    }
+
+    public BodyType getById(int id) {
+        return jdbcTemplate.query("SELECT * FROM body_types WHERE id=?", new BodyTypeMapper(), id)
+                .stream()
+                .findAny()
+                .orElseThrow(() -> new EmptyResultDataAccessException("Could not find record with id " + id, 1));
```

Comment:
at will be thrown before, as I remember

Reply:
When this exception is throwing, my custom message appears (Could not find record with id). 
Does it mean, that orElseThrow method handles that exception?
Or I should use try-catch block?

Reply:
It is redundant actions. 
We can leave only jdbcTemplate.query("SELECT * FROM body_types WHERE id=?", new BodyTypeMapper(), id), set "queryObject", because we know that only one object must be returned.